File: app/src/main/java/com/example/sqlite/room/ChapterEntity.java
package com.example.sqlite.room;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.ColumnInfo;
import androidx.annotation.NonNull;

@Entity(tableName = "chapters")
public class ChapterEntity {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "chapter_id")
    public long chapterId;

    @NonNull
    @ColumnInfo(name = "title")
    public String title;

    @ColumnInfo(name = "description")
    public String description;

    @ColumnInfo(name = "thumbnail_uri")
    public String thumbnailUri;

    @ColumnInfo(name = "order_no")
    public int orderNo;

    @ColumnInfo(name = "level")
    public String level;

    @ColumnInfo(name = "subject")
    public String subject;

    @ColumnInfo(name = "is_active")
    public int isActive = 1;
}

File: app/src/main/java/com/example/sqlite/room/ChapterEntity.kt
package com.example.sqlite.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "chapters")
data class ChapterEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "chapter_id")
    var chapterId: Long = 0,

    @ColumnInfo(name = "title")
    var title: String = "",

    @ColumnInfo(name = "description")
    var description: String? = null,

    @ColumnInfo(name = "thumbnail_uri")
    var thumbnailUri: String? = null,

    @ColumnInfo(name = "order_no")
    var orderNo: Int = 0,

    @ColumnInfo(name = "level")
    var level: String? = null,

    @ColumnInfo(name = "subject")
    var subject: String? = null,

    @ColumnInfo(name = "is_active")
    var isActive: Int = 1
)
